### PR TITLE
Upgrade Next Safe Action

### LINKED ISSUE #62: Upgrade Next-Safe-Action
We are 2 major versions behind. Some code changes will be needed.

### PR DESCRIPTION
**Pull request by joshuasilva414**
## What
- Bump next-safe-action package in web from v5.x to v7.9.3
- Change "next-safe-action/hook" imports to "next-safe-action/hooks"
- Refactor public, authenticated, and admin actions to use validation chaining
- Replace error throwing with returnValidationError
- Replace remove schema validation on actions where the schema is "z.null()"

### CODE REVIEW

File: apps/web/src/actions/teams.ts [hunk 1]
```diff
@@ -1,20 +1,26 @@
...
 import { eq } from "db/drizzle";
 import { revalidatePath } from "next/cache";
 import { getHacker } from "db/functions";
+import { returnValidationErrors } from "next-safe-action";
 
-export const leaveTeam = authenticatedAction(
-	z.null(),
-	async (_, { userId }) => {
+export const leaveTeam = authenticatedAction
+	.outputSchema(
```
Comment (christianhelp) [inline apps/web/src/actions/teams.ts:14]: Not super familiar with the new way of the safe actions, but glanced at the docs and it seems like you use the `outputSchema` to define what the safe action will return and if not provided, it is inferred. Is this specific safe action not properly interpreted or is there another reason?

Comment (joshuasilva414) [reply apps/web/src/actions/teams.ts:14]: Would make that a separate issue tbh

Comment (christianhelp) [reply apps/web/src/actions/teams.ts:14]: Perhaps my question did not come across clearly. I am simply asking **why** this was used. Not whether it was an issue or something.

Comment (joshuasilva414) [reply apps/web/src/actions/teams.ts:14]: Oh yeah I guess it's not strictly necessary. Can be removed if you want

Comment (christianhelp) [reply apps/web/src/actions/teams.ts:14]: Not a big deal. Was just curious what it was doing 👍

**Review by christianhelp — APPROVED** [review]
Great works thanks!

**Pull request merged**